hasher-matcher-actioner/hmalib/common/message_models.py

```diff
@@ -167,3 +172,85 @@ def send_to_queue(self) -> None:
             QueueUrl=config.writebacks_queue_url,
             MessageBody=self.to_aws_json(),
         )
+
+
+@dataclass
+class URLImageSubmissionMessage:
```

Comment:
nit: the fact that the base version of one submission is `...Message` and the other isn't is confusing to me.

Having a `URLImageSubmissionMessage` contain a DTO like  `URLImageSubmission` class
OR
having `S3ImageSubmission`-> `S3ImageSubmissionMessage`
+  `S3ImageSubmissionBatchMessage` -> `S3ImageSubmissionBatchMessages` 
would help imo.

Reply:
I see the confusion. I'll clarify in docstrings.

The issue is that S3ImageSubmission is not a message. It is part of the S3ImageSubmissionBatchMessage. IMO, a message is not a collection of other messages.

Reply:
(again nit:) Right, I don't think you should call `S3ImageSubmission` a message. At the same time URLImageSubmissionMessage is a submission but S3ImageSubmissionBatchMessage is not. What I dislike is half way making a distinction between a message and submission depending on the source with the resulting access pattern:
https://github.com/facebook/ThreatExchange/pull/693/files/76e74290de70346957561f6a210a44c7d75f9c77#diff-dca029e7f27b7a31a5d85e9275abf7d49bc1535026d25e6173e4d4dcbe76cca3R64-R72